scene/realistic-lighting-and-shadows/src/main/java/com/esri/samples/realistic_lighting_and_shadows/RealisticLightingAndShadowsController.java

```diff
@@ -0,0 +1,217 @@
+
+import static java.lang.StrictMath.floor;
+
+public class RealisticLightingAndShadowsController {
+
+  @FXML private SceneView sceneView;
+  @FXML private Label time;
+  @FXML private Slider timeSlider;
+  @FXML private ComboBox<LightingMode> comboBox;
+  private Surface surface;
+  private Calendar calendar;
+  private SimpleDateFormat dateFormat;
+
+  public void initialize() {
+    try {
+
+      // create a scene and add a basemap to it
+      ArcGISScene scene = new ArcGISScene();
+      scene.setBasemap(Basemap.createTopographic());
+
+      // add the scene to the scene view
+      sceneView.setArcGISScene(scene);
+
+      // add a base surface for elevation data to the scene
+      surface = new Surface();
+      ArcGISTiledElevationSource elevationSource = new ArcGISTiledElevationSource("http://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer");
+      surface.getElevationSources().add(elevationSource);
+      scene.setBaseSurface(surface);
+
+      // add a scene layer
+      final String buildings = "http://tiles.arcgis.com/tiles/P3ePLMYs2RVChkJx/arcgis/rest/services/DevA_BuildingShells/SceneServer/layers/0";
+      ArcGISSceneLayer sceneLayer = new ArcGISSceneLayer(buildings);
+      scene.getOperationalLayers().add(sceneLayer);
+
+      // add a camera and initial camera position
+      Camera camera = new Camera(45.54605153789073, -122.69033380511073, 941.0002111233771, 162.58544227544266, 60.0,0.0);
+      sceneView.setViewpointCamera(camera);
+
+      // set atmosphere effect to realistic
+      sceneView.setAtmosphereEffect(AtmosphereEffect.REALISTIC);
+
+      // set a new calendar and add a date and time
+      calendar = new GregorianCalendar(2018, 7, 10, 12, 00, 0);
+      calendar.setTimeZone(TimeZone.getTimeZone("PST"));
+
+      // set the time label on the control panel
+      sceneView.setSunTime(calendar);
+
+      // tidy string to just return date and time (hours and minutes)
+      dateFormat = new SimpleDateFormat("EEE MMM dd HH:mm");
+      dateFormat.setTimeZone(TimeZone.getTimeZone("PST"));
+      String dateAndTimeTidied = dateFormat.format(calendar.getTime());
+
+      // set a label to display the tidied date and time
+      time.setText(dateAndTimeTidied);
+
+      // set the slider to display tick labels as time strings
+      setSliderLabels();
+
+      // set the combo box to display the lighting modes as strings
+      setComboBoxLabels();
+
+      // update the atmosphere effect based on the lighting mode chosen from the combo box
+      comboBox.getSelectionModel().selectedItemProperty().addListener(e -> {
+        sceneView.setSunLighting(comboBox.getSelectionModel().getSelectedItem());
+      });
+
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Set the sun time based on the time from the slider.
+   */
+  @FXML
+  public void changeTimeOfDay() {
```

Comment:
I personally thing 'update' is a bit more fitting for a method name

```suggestion
  public void updateTimeOfDay() {
```